### PR TITLE
Fixes #9076: Forbid agent reinit on the root server

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9076

### CODE REVIEW

File: share/commands/agent-reinit [hunk 1]
```diff
@@ -8,6 +8,11 @@
 # @man *WARNING*: This command will permanently delete your node uuid and keys, and no configuration will
 # @man be applied before re-accepting and configuring the node on the server.
 
+if [ $(cat /opt/rudder/etc/uuid.hive 2>/dev/null) = "root" ]
+then
+  echo "Reinitializing the root server is forbiden" >&2
```
Comment (amousset) [inline share/commands/agent-reinit]: We should explain quickly why.

File: share/commands/agent-reset [hunk 2]
```diff
@@ -6,9 +6,12 @@
 # @man without caching interference or when you have trouble with the promises updates.
 
 # Try to remove everything that can block
-# - restore initial promises
-rm -rf /var/rudder/cfengine-community/inputs/*
-cp -r /opt/rudder/share/initial-promises/* /var/rudder/cfengine-community/inputs
+# - restore initial promises except on root
```
Comment (amousset) [inline share/commands/agent-reset]: Could you add a note in the man saying it won't reset policies on root servers?

File: share/commands/agent-reinit (hunk 1 above)
Comment (jooooooon) [inline share/commands/agent-reinit]: Forbidden*

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by jooooooon** [comment]
@amousset We need you to approve the changes you requested to move forwards

**Comment by jooooooon** [comment]
Ping @peckpeck - this needs merging manually

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**